src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataDeltaFeedSerializer.cs

```diff
@@ -441,12 +489,20 @@ public override async Task WriteObjectAsync(object graph, Type type, ODataMessag
         /// <param name="writeContext">The <see cref="ODataSerializerContext"/>.</param>
         public virtual void WriteDeltaDeletedEntry(object graph, ODataWriter writer, ODataSerializerContext writeContext)
         {
-            ODataDeletedResource deletedResource = GetDeletedResource(graph);
+            ODataResourceSerializer serializer = SerializerProvider.GetEdmTypeSerializer(_elementType) as ODataResourceSerializer;
+            ResourceContext resourceContext = serializer.GetResourceContext(graph, writeContext);
+            SelectExpandNode selectExpandNode = serializer.CreateSelectExpandNode(resourceContext);
 
-            if (deletedResource != null)
+            if (selectExpandNode != null)
```

Comment:
Could you explain this change?

Reply:
we've added a null check for `selectExpandNode` before calling the code that was there before. This is because we've added  this method call: 

```c#
serializer.WriteDeltaComplexProperties(selectExpandNode, resourceContext, writer);
```